### DISCUSSION TITLE
Empty Task Groups Appear in Graph View

### DISCUSSION THREAD

**Discussion by MatrixManAtYrService** [apache/airflow#26741] [Ideas]
### Apache Airflow version

2.4.0

### What happened

This is not a regression, it's just that I comes up so rarely that I hadn't bothered with the bug until now.  Here's two dags:

```python3
from airflow import DAG
from airflow.decorators import task, task_group
from airflow.operators.empty import EmptyOperator
from airflow.operators.python import PythonOperator
from datetime import datetime

things_one = [1, 2, 3]
things_two = []

@task_group
def fan_out(upstream_task, the_things):
    tasks = []
    for thing in the_things:
        task = EmptyOperator(task_id=str(thing))
        upstream_task >> task
        tasks.append(task)
    return tasks or upstream_task

with DAG(dag_id="one", start_date=datetime(1970,1,1), schedule_interval=None) as dag:
    start = EmptyOperator(task_id="start")
    end = EmptyOperator(task_id="end")
    fan_out(start, things_one) >> end

with DAG(dag_id="two", start_date=datetime(1970,1,1), schedule_interval=None) as dag:
    start = EmptyOperator(task_id="start")
    end = EmptyOperator(task_id="end")
    fan_out(start, things_two) >> end
```

And the corresponding graph views:

<img width="324" alt="Screen Shot 2022-09-26 at 8 02 58 PM" src="https://user-images.githubusercontent.com/5834582/192414701-3bcd7410-ec69-4120-9a1c-11c39d0328a9.png">

<img width="276" alt="Screen Shot 2022-09-26 at 8 03 15 PM" src="https://user-images.githubusercontent.com/5834582/192414724-e7b672cb-f996-4b03-a654-78a5fa96aebc.png">

Note that there are no tasks in the `fan_out` task group in the `two` dag, yet a disconnected task group shows up.

### What you think should happen instead

If a TaskGroup has no tasks, it should not be shown.

### How to reproduce

View the graph view for the above dags file

### Operating System

docker/debian

### Versions of Apache Airflow Providers

n/a

### Deployment

Astronomer

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by eladkal** [comment]
> If a TaskGroup has no tasks, it should not be shown

Why? User did code it.

**Comment by MatrixManAtYrService** [comment]
@eladkal ok, that's fair. If you think that somebody wants this behavior I'll happily close the ticket.  I had assumed that it was the way it was but just because we didn't give it much thought.

I dislike it because when people look at my DAG and see a box floating out of context, they assume that it's of elevated importance and not diminished importance.  

They ask:
> What does "process mutations"  mean?

To which I answer:
> It means that we don't have any mutations to process. If we did, they'd appear here... 

I can work around it by adding an EmptyOperator to my task groups so there's something anchor them to the rest of the dag even when they're generated by a loop or condition that happens to be empty this time.

**Comment by eladkal** [comment]
> I dislike it because when people look at my DAG and see a box floating out of context

True but on the other hand in complex dags where you have dozens of tasks and task group you dont want the UI to hide stuff it will be harder to identify and remove redundant code.

Could we maybe mark in the UI that its empty? Maybe diffrent edges or something like that?

**Comment by MatrixManAtYrService** [comment]
Making it look different would at least prevent it from looking like something that I should click to expand.

It's such an unlikely case though, seems weird to dedicate a style to it. Another approach would be to have it say 
> fan_out
>(0 tasks)

Honestly though, I hadn't thought this deeply when I wrote the issue. My gut reaction was:

> well that's obviously a bug

So I wrote an issue. If my gut was miscalibrated in this case, well we probably have bigger fish to fry and I'm happy to just close it.

**Comment by jedcunningham** [comment]
I also think we should still show it 🤷‍♂️. In general I'm not a fan of just hiding things.

**Comment by eladkal** [comment]
OK so I'm converting this one to discussion as an idea :)
